### DISCUSSION TITLE
Package size

### DISCUSSION THREAD

**Discussion by mbojan** [wlandau/instantiate#22] [General]
I'm developing https://github.com/coalesce-lab/stansum and I noticed that the installed package takes-up quite some disk space on Linux (286MB), even if ZIPped (59MB) -- which is a good estimate of how large the Windows binary version of the package will be. I did not submit the package to CRAN yet, but I foresee this to be an issue. CRAN does not like big packages.

For comparison, CRAN says that **instantiate** has at this moment (2024-05-27) a single reverse-dependency which is **hdbayes** (https://github.com/ethan-alt/hdbayes). Installed it takes-up 28MB and 9MB when ZIPped.  That's still substantial size, although considerably smaller than **stansum**. In terms of content **hdbayes** wraps 9 Stan models and **stansum** 13 -- not a big difference.

I'm wondering:

- Where's the size difference coming from?
- Is there anything that can be done to make package leaner?

**Comment by wlandau** [comment]
It's probably because of the complied Stan models. Stan is large and creates large model files. Those models do not automatically compile on CRAN because CRAN does not have CmdStan, so package size should not interfere with CRAN submissions.

**Reply by mbojan** [reply]
> Those models do not automatically compile on CRAN because CRAN does not have CmdStan, so package size should not interfere with CRAN submissions.

Ha, thanks! I didn't realize that.
